caffe2/core/tensor.h

```diff
@@ -102,6 +102,25 @@ class CAFFE2_API Tensor final {
     return x;
   }
 
+  /**
+   * Clone self as a Tensor that share the same Storage
```

Comment:
I'd like this doc to be beefier. What does it mean to "share a same storage"? If I subsequently resize one of the aliases of the tensor, what happens? What if I restride it in-place?